internal/app/server/server.go

```diff
@@ -15,32 +15,68 @@
 package server
 
 import (
+	"bufio"
 	"context"
 	"errors"
-	"flag"
 	"fmt"
 	"io"
+	"net"
 	"net/http"
-	"os"
-	"os/signal"
-	"syscall"
-	"time"
 
 	"connectrpc.com/conformance/internal/gen/proto/connect/connectrpc/conformance/v1alpha1/conformancev1alpha1connect"
+	v1alpha1 "connectrpc.com/conformance/internal/gen/proto/go/connectrpc/conformance/v1alpha1"
 	"github.com/rs/cors"
 	"golang.org/x/net/http2"
 	"golang.org/x/net/http2/h2c"
+	"google.golang.org/protobuf/encoding/protojson"
+	proto "google.golang.org/protobuf/proto"
+)
+
+const (
+	// The default host to use for the server
+	defaultHost = "127.0.0.1"
+	// The default port to use for the server. We choose 0 so that
+	// an ephemeral port is selected by the OS
+	defaultPort = "0"
 )
 
 func Run(ctx context.Context, args []string, in io.ReadCloser, out, err io.WriteCloser) error {
-	// TODO - The ports should not be given to the run call. Instead the server should read a
-	// ServerCompatRequest and start itself. It should then return the port numbers back on stdout
-	// as a ServerCompatResponse
-	h1Port := flag.String("h1Port", "8080", "port for HTTP/1.1 traffic")
-	h2Port := flag.String("h2Port", "8081", "port for HTTP/2 traffic")
+	rdr := bufio.NewReader(in)
+
+	var data string
+	// TODO - How should we read from 'in'? Using ReadString or ReadBytes?
+	// Assuming ReadBytes, but I used ReadString so i could test by redirecting a file
```

Comment:
Since we will typically be reading binary data, we should read bytes, not a string 🙂 

You can simply use `io.ReadAll` here if I'm not mistaken.